### DISCUSSION TITLE
Support FileMaker Cloud

### DISCUSSION THREAD

**Discussion by heebinho** [fuzzzerd/fmdata#195] [Ideas]
Hi Nate,

have you ever thought about supporting FileMaker Cloud? The process to obtain a token would be slightly different:

1. The first step would be to get the "AWS Cognito FileMaker Cloud UserPool id"
`GET https://www.ifmcloud.com/endpoint/userpool/2.2.0.my.claris.com.json`

2. The second step would be to get the token as showed in this [gist](https://gist.github.com/heebinho/49f17739c97fb938b5abfd8f529e511c)

3. Then the authorization header would have to be set like this:
'Authorization FMID eyJ...'
instead of the base 64 encoded username:password
`            // special non-token auth to list databases
            requestMessage.Headers.Authorization = new AuthenticationHeaderValue("basic", Convert.ToBase64String(
                    Encoding.UTF8.GetBytes($"{_userName}:{_password}")`

Awesome project, already learned a lot.

Greetings from Zurich

**Comment by heebinho** [comment]
I got a first draft - see [commit](https://github.com/heebinho/fmdata/commit/1e3caa6ce2c03d6ba0898fd41c7c9a110c35eb06)
I guess this won't make up for a pull request, but it's a working version to show the AWS cognito authorization flow.

**Reply by fuzzzerd** [reply]
That's a very interesting idea. I'm interested to learn more about how it could work and figure out a way to get it working.

Couple of initial thoughts.

1. We don't want to add the aws packages to the base package
2. We need to abstract authentication such that another package can handle the aws part and adjust the main httpclient.
3. I'm not opposed to adding more targets, but I'd be more interested in net6.0, or maybe net48 than net47.

**Reply by heebinho** [reply]
Glad to hear and I totally agree with your first two points.
As for three, I'm with you - net6.0 support would be great. (net47 was just for testing purposes on an older windows server)

**Reply by fuzzzerd** [reply]
I do not have access to FileMaker Cloud to test against, and I've got some specific guidelines on what would be necessary for me to accept a PR for this functionality, but if you're game I'd be happy to work with you on getting something working and eventually merged.

**Comment by fuzzzerd** [comment]
This has been implemented in #213.

**Reply by fuzzzerd** [reply]
This has been released to Nuget in v5!

**Reply by heebinho** [reply]
Thank you Nate!!
